robocop/checkers/naming.py

```diff
@@ -726,61 +759,168 @@ def visit_KeywordCall(self, node):  # noqa
 class SimilarVariableChecker(VisitorChecker):
     """Checker for finding same variables with similar names."""
 
-    reports = ("possible-variable-overwriting",)
+    reports = ("possible-variable-overwriting", "inconsistent-variable-name")
 
     def __init__(self):
-        self.variables = defaultdict(set)
+        self.assign_variables = defaultdict(list)
```

Comment:
I'm not a fan of this name. It suggests an activity, because `assign` is used like a verb here. Maybe we should name it `variable_variants`?

Reply:
I agree on name suggesting activity. It's strictly containing variables that were assigned somewhere (from keyword calls, for loop helper variable, if return). The ``variable_variants`` would be too generic. Maybe ``assigned_variables``? If not, I will go with ``variable_variants

Reply:
`assigned_variables` is fine, let's do it.